### PR TITLE
Feature/15 request endpoint

### PR DESCRIPTION
**Pull request by sjongejan**

### CODE REVIEW

File: app/Http/Controllers/Api/RequestApiController.php [hunk 1]
```diff
@@ -3,53 +3,63 @@
...
+     * Dispatch ExecuteRequestJob.
+     *
+     * @param Request $request
+     * @return Response
+     */
+    public function storeRequest(Request $request): Response
     {
-        return response()->json([
-            'created' => ($request->input('this') == 'works'),
+        $this->validate($request, [
```
Comment (FaaPz) [inline app/Http/Controllers/Api/RequestApiController.php:51]: Doesn't this work? 👉  https://laravel-news.com/request-data-validator-improvements

File: app/Parsers/RequestHeaderParser/RequestHeaderParser.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    /**
+     * Get status code by HTTP reason phrase.
+     *
+     * @param $reasonPhrase
+     * @return int
+     */
+    private function getStatusCode($reasonPhrase): int
+    {
+        preg_match('/[0-9]{3}/', $reasonPhrase, $matches);
```
Comment (FaaPz) [inline app/Parsers/RequestHeaderParser/RequestHeaderParser.php]: This regular expression should be stricter. Please only accept `1-5` as first character/number.

File: app/Parsers/RequestHeaderParser/RequestHeaderParser.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+     * Execute request and parse responses with headers.
+     *
+     * @return Response[]
+     */
+    public function getResponses(): array
+    {
+        $this->execute();
+
+        /* @var Response[] $responses */
+        $responses = [];
```
Comment (FaaPz) [inline app/Parsers/RequestHeaderParser/RequestHeaderParser.php:85]: Why not using a `Collection`?

File: app/Parsers/RequestHeaderParser/RequestHeaderParser.php [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+    /**
+     * Execute HTTP request to url with request method.
+     *
+     * @throws FailedFetchingResponseHeadersException
+     */
+    private function execute()
+    {
+        if (! $this->method || ! $this->url) {
+            return;
```
Comment (FaaPz) [inline app/Parsers/RequestHeaderParser/RequestHeaderParser.php]: Maybe throwing an exception, or not? :thinking:

**Review by FaaPz — COMMENTED** [review]
I left some comments. Even though we wrote this together (pair programming FTW! 🎉 ), I think we should at least consider some of my comments.

File: app/Http/Controllers/Api/RequestApiController.php (hunk 1 above)
Comment (sjongejan) [reply app/Http/Controllers/Api/RequestApiController.php:51]: Doesn't work in Lumen :(

File: app/Parsers/RequestHeaderParser/RequestHeaderParser.php (hunk 3 above)
Comment (sjongejan) [reply app/Parsers/RequestHeaderParser/RequestHeaderParser.php:85]: The RequestHeaderParser is a class that can be used in multiple projects/environment/packages. It should be nice if collections are not mandatory.

File: app/Http/Controllers/Api/RequestApiController.php (hunk 1 above)
Comment (FaaPz) [reply app/Http/Controllers/Api/RequestApiController.php:51]: Too bad... 😕

**Review by FaaPz — APPROVED** [review]
Let's go with this! 👍

**Pull request merged**